src/function_4.py

```
@@ -0,0 +1,104 @@
+    for (x, y, w, h) in faces:
+        roi_color = frame[y:y+h, x:x+w]
+
+        # Nếu Training có preprocessing → Realtime cũng phải preprocess
+        if enable_optional_preprocess:
+            roi_processed = preprocess_face_image(roi_color)
+        else:
+            roi_processed = cv2.cvtColor(roi_color, cv2.COLOR_BGR2GRAY)
+
+        try:
+            label_id, confidence = recognizer.predict(roi_processed)
+        except:
```

Flake8 (7.2.0)

[error] do not use bare 'except'

(E722)